File: src/main/java/im/tox/tox4j/core/callbacks/FriendReadReceiptCallback.java
package im.tox.tox4j.core.callbacks;

/**
 * This event is triggered when the friend receives the message sent with
 * {@link im.tox.tox4j.core.ToxCore#sendMessage} with the corresponding message ID.
 */
public interface FriendReadReceiptCallback {
  FriendReadReceiptCallback IGNORE = new FriendReadReceiptCallback() {
    @Override
    public void friendReadReceipt(int friendNumber, int messageId) {
    }
  };

  /**
   * @param friendNumber The friend number of the friend who received the message.
   * @param messageId The message ID as returned from {@link im.tox.tox4j.core.ToxCore#sendMessage} corresponding to the
   *                  message sent.
   */
  void friendReadReceipt(int friendNumber, int messageId);
}

File: projects/tox4j/src/main/java/im/tox/tox4j/core/callbacks/FriendReadReceiptCallback.scala
package im.tox.tox4j.core.callbacks

import im.tox.tox4j.core.ToxCore

/**
 * This event is triggered when the friend receives the message sent with
 * [[ToxCore.friendSendMessage]] with the corresponding message ID.
 */
trait FriendReadReceiptCallback[ToxCoreState] {
  /**
   * @param friendNumber The friend number of the friend who received the message.
   * @param messageId The message ID as returned from [[ToxCore.friendSendMessage]] corresponding to the message sent.
   */
  def friendReadReceipt(
    friendNumber: Int, messageId: Int
  )(state: ToxCoreState): ToxCoreState = state
}
